Userland/Libraries/LibJS/Runtime/Value.cpp

```diff
@@ -496,6 +553,16 @@ ThrowCompletionOr<Value> Value::to_number(GlobalObject& global_object) const
             return js_infinity();
         if (string == "-Infinity")
             return js_negative_infinity();
+
+        auto result = parse_number_text(string);
+        if (result.has_value()) {
+            if (result->invalid)
+                return js_nan();
+
+            auto bigint = Crypto::UnsignedBigInteger::from_base(result->base, result->literal);
+            return Value(bigint.to_double());
```

Comment:
Going through a UnsignedBigInteger seems overkill here and `to_double` is naive.
For example `Number("0xfffffffffffff000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000");` is valid but will not work with this.

To be honest this is actually quite hard so we can probably ignore that case, but just using `strtoul(string_start, string_end, base)` to convert the base

Reply:
Ah sorry about that, I was using big integer as I was originally aiming to have it more closely match the behavior of other js engines (returning numbers bigger than the 64bit integer limit), but I realized later on that the behavior I was going for isn't supported elsewhere in the engine so it wouldn't have made sense for just this one thing to support said behavior.

Reply:
We should support this behavior (eventually anyway) but since `to_double` right now does not support it we probably shouldn't use it

Reply:
I have now implement correct (mostly) `to_double` behavior in https://github.com/SerenityOS/serenity/pull/14929 so feel free to go back to using `(Un)signedBigInteger` with `to_double`